editor/src/main/java/io/github/rosemoe/sora/widget/EditorKeyEventHandler.java

```diff
@@ -359,6 +359,8 @@ private Boolean handleKeyEvent(KeyEvent event,
                         } else {
                             editor.getSnippetController().shiftToNextTabStop();
                         }
+                    } if (editor.getProps().indentSelectionWithTab && editorCursor.isSelected()) {
```

Comment:
If selected text is on a single line, the key event should still be intepreted as replacing the selected text to a single TAB.

Reply:
I actually tested the feature in vscode and it behaves like what I've described.

In that case, maybe current implementation is better. The change and toggle is unnecessary. Sorry.